src/ai_hawk/job_manager.py

```diff
@@ -484,15 +490,27 @@ def extract_job_information_from_tile(self, job_tile):
 
     def is_blacklisted(self, job_title, company, link, job_location):
         logger.debug(f"Checking if job is blacklisted: {job_title} at {company} in {job_location}")
-        job_title_words = job_title.lower().split(' ')
-        title_blacklisted = any(word in job_title_words for word in map(str.lower, self.title_blacklist))
-        company_blacklisted = company.strip().lower() in (word.strip().lower() for word in self.company_blacklist)
-        location_blacklisted= job_location.strip().lower() in (word.strip().lower() for word in self.location_blacklist)
+        title_blacklisted = any(re.search(pattern, job_title, re.IGNORECASE) for pattern in self.title_blacklist_patterns)
+        company_blacklisted = any(re.search(pattern, company, re.IGNORECASE) for pattern in self.company_blacklist_patterns)
+        location_blacklisted = any(re.search(pattern, job_location, re.IGNORECASE) for pattern in self.location_blacklist_patterns)
         link_seen = link in self.seen_jobs
         is_blacklisted = title_blacklisted or company_blacklisted or location_blacklisted or link_seen
         logger.debug(f"Job blacklisted status: {is_blacklisted}")
 
-        return title_blacklisted or company_blacklisted or location_blacklisted or link_seen
+        return is_blacklisted
+
+    def generate_regex_patterns(self, blacklist):
+        # Converts each blacklist entry to a regex pattern that ensures all words appear, in any order
+        patterns = []
+        for term in blacklist:
+            # Split term into individual words
+            words = term.split()
+            # Create a lookahead for each word to ensure it appears independently
+            lookaheads = [fr"(?=.*\b{re.escape(word)}\b)" for word in words]
+            # Combine lookaheads with a pattern that allows flexible separators between the words
+            pattern = "".join(lookaheads)  # Ensures all words are present
```

Comment:
can you share few examples of final pattern looks like

Reply:
of course. should i add them in code as comments as well?

Example of pattern for job title:
title_blacklist = ["Data Engineer", "Software Engineer"]
patterns = ['(?=.*\\bData\\b)(?=.*\\bEngineer\\b)', '(?=.*\\bSoftware\\b)(?=.*\\bEngineer\\b)']
     
'?=.*' => Regex expression that allows us to check if the following pattern appears somewhere in the string searched, even if there are any characters before the word, like '... , Data'.
 '\b{WORD}\b' => Regex expression for a word boundry, that the WORD is treated as whole words                  rather than as parts of other words.

In this example, it will match for:
"Engineer, Data", "Engineer of Data", "Senior Data - Engineer".
But wont match for:
"Database Engineer", as Data is treated as a whole word
"Project Engineer", as it must have both Data and Engineer because of the lookahead.